spec/front_desk_spec.rb

```diff
@@ -0,0 +1,398 @@
+        expect(@concierge.reservations_record[0].room_blocks[4]).must_equal 5
+        expect(@concierge.reservations_record[0].room_blocks.length).must_equal 5
+        expect(reservation_1.check_in).must_equal Date.new(2019, 3, 20)
+        expect(reservation_1.check_out).must_equal Date.new(2019, 3, 24)
+      end
+    end
+
+    describe "get reservation by a specific date" do
+      before do
+        @concierge = build_front_desk
+      end
+
+      it "will get a reservation by a date" do
+        reservation_2 = @concierge.make_reservation(
+          1202,
+          2,
+          9,
+          Date.new(2019, 4, 8),
+          Date.new(2019, 4, 10),
+        )
+
+        @concierge.add_reservation(reservation_2)
+
+        by_date = @concierge.get_reservations_by_date(Date.new(2019, 4, 8))
+
+        expect(by_date).must_be_kind_of Array
+        expect(by_date.length).must_equal 1
+        expect(reservation_2.check_in).must_be :<, Date.new(2019, 4, 10)
+        expect(reservation_2.check_in).must_be :>, Date.new(2019, 4, 7)
+      end # it
+    end # describe
+
+    describe "room_availability method" do
+      before do
+        @concierge = build_front_desk
+      end # before
+
+      it "will return array of available rooms" do
+        reservation_3 = @concierge.make_reservation(
+          1001,
+          3,
+          15,
+          Date.new(2019, 4, 8),
+          Date.new(2019, 4, 21),
+        )
+        @concierge.add_reservation(reservation_3)
+
+        all_available_rooms = @concierge.room_availability(
+          Date.new(2019, 4, 8)
+        )
+
+        expect(all_available_rooms).must_be_kind_of Array
+        expect(all_available_rooms.length).must_equal 17
+      end # it
+    end # describe
+
+    describe "raise an ArgumentError if check_in date is before current date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an error if check_in date is before current date" do
+        expect {
+          @concierge.make_reservation(
+            30011,
+            1,
+            19,
+            Date.new(2019, 3, 10),
+            Date.new(2019, 3, 25)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "raise an ArgumentError if check_out date is before check_in date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an error if check_out date is before check_in date" do
+        expect {
+          @concierge.make_reservation(
+            30111,
+            1,
+            20,
+            Date.new(2019, 3, 28),
+            Date.new(2019, 3, 25)
+          )
+        }.must_raise ArgumentError
+      end
+    end
+
+    describe "raise an ArgumentError if check_out date is before current date" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will raise an ArgumentError if check_out date is before current date" do
```

Comment:
The logic for this test ends up living in the `Reservation` class, and you even test it there -- no need to double test it here in this file, unless this code was doing something different